gabbi/tests/test_runner.py

```diff
@@ -14,31 +14,61 @@
 """
 
 from io import StringIO
+import os
+import socket
+import subprocess
 import sys
+import time
 import unittest
-from uuid import uuid4
-
-from wsgi_intercept.interceptor import Urllib3Interceptor
 
 from gabbi import exception
 from gabbi.handlers import base
 from gabbi.handlers.jsonhandler import JSONHandler
 from gabbi import runner
-from gabbi.tests.simple_wsgi import SimpleWsgi
+
+
+def get_free_port():
+    sock = socket.socket()
+    sock.bind(('', 0))
+    return sock.getsockname()[1]
+
+
+class ForkedWSGIServer:
+
+    def __init__(self, host, port):
+        self.host = host
+        self.port = port
+
+    def start(self):
+        self.process = subprocess.Popen(
+            [
+                "python",
+                "gabbi/tests/external_server.py",
+                self.host,
+                str(self.port)
+            ],
+            env=os.environ.update({"PYTHONPATH": "."}),
+            close_fds=True,
+        )
+        # We need to sleep a bit to let the wsgi server start.
+        # TODO(cdent): This is regrettable.
+        time.sleep(.4)
```

Comment:
Can we not poll the server until it responds?

Reply:
I went for the simplest thing. If it becomes an issue, is fixable.